lib/membrane_audio_mixer.ex

```diff
@@ -97,345 +98,238 @@ defmodule Membrane.AudioMixer do
-      end
-
-    {actions, state} = mix_and_get_actions(context, state)
-
+  defp do_handle_demand(target_queues_size, state) do
     actions =
-      if all_streams_ended?(context) do
-        actions ++ [{:end_of_stream, :output}]
-      else
-        actions
-      end
+      state.pads_data
+      |> Enum.map(fn {pad, %{queue: queue}} ->
+        demand = max(0, target_queues_size - byte_size(queue))
+        {:demand, {pad, demand}}
+      end)
 
     {actions, state}
   end
 
   @impl true
-  def handle_event(pad, event, _context, state) do
+  def handle_event(pad, event, _ctx, state) do
     Membrane.Logger.debug("Received event #{inspect(event)} on pad #{inspect(pad)}")
-
     {[], state}
   end
 
   @impl true
-  def handle_buffer(
-        pad_ref,
-        buffer,
-        context,
-        state
-      ) do
-    ready_to_mix? = get_in(state.pads, [pad_ref, :ready_to_mix?])
-    do_handle_buffer(pad_ref, buffer, ready_to_mix?, context, state)
-  end
-
-  defp do_handle_buffer(
-         pad_ref,
-         %Buffer{payload: payload, pts: pts},
-         false,
-         context,
-         %{stream_format: stream_format, pads: pads, last_ts_sent: last_ts_sent} = state
-       ) do
-    offset = get_in(pads, [pad_ref, :offset])
-    buffer_ts = pts + offset
-
-    state =
-      if buffer_ts >= last_ts_sent do
-        diff = buffer_ts - last_ts_sent
-        silence = RawAudio.silence(stream_format, diff)
-
-        update_in(
-          state,
-          [:pads, pad_ref],
-          &%{&1 | queue: silence <> payload, ready_to_mix?: true}
-        )
-      else
-        state
+  def handle_stream_format(pad, stream_format, ctx, state) do
+    {actions, state} =
+      cond do
+        ctx.pads.output.stream_format != nil -> {[], state}
+        state.stream_format != nil -> set_stream_format(state.stream_format, state)
+        true -> set_stream_format(stream_format, state)
       end
 
-    size = byte_size(get_in(state, [:pads, pad_ref, :queue]))
-    time_frame = RawAudio.frame_size(stream_format)
+    :ok = validate_input_stream_format!(pad, stream_format, state)
 
-    mix_and_redemand(size, time_frame, context, state)
+    {actions, state}
   end
 
-  defp do_handle_buffer(
-         pad_ref,
-         %Buffer{payload: payload},
-         true,
-         context,
-         %{stream_format: stream_format, pads: pads} = state
-       ) do
-    {size, pads} =
-      Map.get_and_update(
-        pads,
-        pad_ref,
-        fn %{queue: queue} = pad ->
-          {byte_size(queue) + byte_size(payload), %{pad | queue: queue <> payload}}
-        end
-      )
-
-    time_frame = RawAudio.frame_size(stream_format)
-    mix_and_redemand(size, time_frame, context, %{state | pads: pads})
+  defp set_stream_format(%RemoteStream{}, _state) do
+    raise """
+    You need to specify `stream_format` in options if `Membrane.RemoteStream` will be received on the `:input` \
+    pad and you cannot pas `Membrane.RemoteStream{}` in this option.
```

Comment:
```suggestion
    You need to specify `stream_format` in options if `Membrane.RemoteStream` will be received on the `:input` pad and you cannot pass `%Membrane.RemoteStream{}` in this option.
```

Reply:
IMO so wide lines are hard to read on narrow screens and it is not very aesthetic to have strings significantly wider than rest of the code